pkg/operator/configobservation/scheduler/observe_scheduler.go

```diff
@@ -39,30 +40,73 @@ func ObserveSchedulerConfig(genericListers configobserver.Listers, recorder even
-	case len(configMapName) > 0:
-		sourceTargetLocation = resourcesynccontroller.ResourceLocation{
-			Namespace: operatorclient.GlobalUserSpecifiedConfigNamespace,
-			Name:      configMapName,
+	// Deprecated handling for Policy config
+	// TODO(@damemi): Remove this when Policy API is removed
+	if len(schedulerConfig.Spec.Policy.Name) > 0 {
+		configMapName := schedulerConfig.Spec.Policy.Name
+
+		switch {
+		case len(configMapName) == 0:
+			sourceTargetLocation = resourcesynccontroller.ResourceLocation{}
+		case len(configMapName) > 0:
+			sourceTargetLocation = resourcesynccontroller.ResourceLocation{
+				Namespace: operatorclient.GlobalUserSpecifiedConfigNamespace,
+				Name:      configMapName,
+			}
+		}
+
+		// Sync the configmap from openshift-config namespace to openshift-kube-scheduler namespace. If the configMapName
+		// is empty string, it will mirror the deletion as well.
+		err = listers.ResourceSyncer().SyncConfigMap(
+			resourcesynccontroller.ResourceLocation{
+				Namespace: operatorclient.TargetNamespace,
+				Name:      "policy-configmap",
+			},
+			sourceTargetLocation,
+		)
+		if err != nil {
+			errs = append(errs, err)
+			return prevObservedConfig, errs
+		}
+		return observedConfig, errs
+	}
+
+	currentProfiles, _, err := unstructured.NestedSlice(existingConfig, "profiles")
+	if err != nil {
+		return prevObservedConfig, append(errs, err)
+	}
+	if len(currentProfiles) > 0 {
+		if err := unstructured.SetNestedField(prevObservedConfig, currentProfiles, "profiles"); err != nil {
+			errs = append(errs, err)
 		}
 	}
 
-	// Sync the configmap from openshift-config namespace to openshift-kube-scheduler namespace. If the configMapName
-	// is empty string, it will mirror the deletion as well.
-	err = listers.ResourceSyncer().SyncConfigMap(
-		resourcesynccontroller.ResourceLocation{
-			Namespace: operatorclient.TargetNamespace,
-			Name:      "policy-configmap",
-		},
-		sourceTargetLocation,
-	)
+	if len(schedulerConfig.Spec.Profiles.Name) == 0 {
+		if len(currentProfiles) > 0 {
+			recorder.Eventf("ObservedConfigMapNameChanged", "scheduler profiles configmap removed")
+			unstructured.RemoveNestedField(prevObservedConfig, "profiles")
+		}
+		return prevObservedConfig, errs
+	}
+	profileConfig, err := listers.ConfigmapLister.ConfigMaps(operatorclient.GlobalUserSpecifiedConfigNamespace).Get(schedulerConfig.Spec.Profiles.Name)
 	if err != nil {
 		errs = append(errs, err)
 		return prevObservedConfig, errs
 	}
+	config := profileConfig.Data["profiles.cfg"]
+	if len(config) == 0 && len(currentProfiles) > 0 {
+		recorder.Eventf("ObservedProfilesChanged", "scheduler profiles removed")
+		return prevObservedConfig, errs
+	}
+	profiles := map[string][]interface{}{}
+	if err := yaml.Unmarshal([]byte(config), &profiles); err != nil {
+		errs = append(errs, err)
+		return prevObservedConfig, errs
+	}
+	if err := unstructured.SetNestedSlice(observedConfig, profiles["profiles"], "profiles"); err != nil {
```

Comment:
Worth noting `observedConfig` here corresponds to `KubeSchedulerConfiguration` data type which is rendered into `config.yaml` filed passed to kube-scheduler binary.